src/workflows/registration.ts

```diff
@@ -113,3 +116,19 @@ export async function getSignableRegistrationOnchain(
     payload_hash: response.data.payload_hash,
   };
 }
+
+async function isUserRegistered(
+  userAddress: string,
+  usersApi: UsersApi,
+): Promise<boolean> {
+  try {
+    await usersApi.getUsers({ user: userAddress });
```

Comment:
if there's no matched user, what's the API would return? 
I am not sure if we need to return
```
return !!userAPI.getUser2({ user: userAddress }) 
```

Reply:
If the user is not registered, the function `getUsers` throws an AxiosError with status 404, otherwise `getUsers` will return an AxiosResponse with `'accounts': Array<string>`.